internal/executor/job/job_context.go

```diff
@@ -276,6 +277,21 @@ func (c *ClusterJobContext) handleDeletedPod(pod *v1.Pod) {
 	if jobId != "" {
 		isUnexpectedDeletion := !util.IsMarkedForDeletion(pod) && !util.IsPodFinishedAndReported(pod)
 		if isUnexpectedDeletion {
+			// Preempted cluster event occurs after the pod gets killed (preempted)
+			// Wait for 3 seconds so the Preempted event gets recorded
+			time.Sleep(3 * time.Second)
```

Comment:
What effect does sleeping for 3 seconds here have?  Does it block anything else?  

I guess I'm thinking about the case where a whole bunch of pods are pre-empted.  Doe this lead to the executor pausing processing events for any length of time?

Reply:
`	// AddEventHandler adds an event handler to the shared informer using the shared informer's resync
	// period.  Events to a single handler are delivered sequentially, but there is no coordination
	// between different handlers.
	AddEventHandler(handler ResourceEventHandler)`

I guess I should wrap this into a goroutine